From ab3cdae19cefdb925fdd67bbd56618b7522e6e04 Mon Sep 17 00:00:00 2001
From: Bogdan Roman <bogdan-gabriel.roman@nxp.com>
Date: Tue, 12 Apr 2022 10:56:59 +0300
Subject: [PATCH 097/103] s32g: use DM_PMIC only for s32g-evb/rdb platforms

The VR5510 PMIC is present on S32G3EVB and S32GRDB platforms. Use the
driver only for them.

Issue: ALB-8597
Upstream-Status: Pending 

Signed-off-by: Bogdan Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/mach-s32/Kconfig     | 2 --
 board/nxp/s32-cc/s32g/Kconfig | 4 ++++
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-s32/Kconfig b/arch/arm/mach-s32/Kconfig
index fc010061f0..3b294646bd 100644
--- a/arch/arm/mach-s32/Kconfig
+++ b/arch/arm/mach-s32/Kconfig
@@ -20,8 +20,6 @@ config NXP_S32CC
 	imply DM_ETH
 	imply DM_I2C
 	imply DM_PCI
-	imply DM_PMIC
-	imply DM_PMIC_VR5510
 	imply DM_SPI
 	imply DM_SPI_FLASH
 	imply DWC_ETH_QOS_DEVICES
diff --git a/board/nxp/s32-cc/s32g/Kconfig b/board/nxp/s32-cc/s32g/Kconfig
index 8747c6e4fa..d55cd5f4ae 100644
--- a/board/nxp/s32-cc/s32g/Kconfig
+++ b/board/nxp/s32-cc/s32g/Kconfig
@@ -1,6 +1,8 @@
 config NXP_S32GRDB_BOARD
 	bool
 	default n
+	imply DM_PMIC
+	imply DM_PMIC_VR5510
 	imply NVME
 	select ADC
 	select BOARD_LATE_INIT
@@ -10,6 +12,8 @@ config NXP_S32GEVB_BOARD
 	bool
 	default n
 	imply ADC
+	imply DM_PMIC
+	imply DM_PMIC_VR5510
 	imply SARADC_S32CC
 
 config S32GEN1_HWCONFIG
-- 
2.25.1

